client/network/sync/src/warp.rs

```diff
@@ -70,21 +80,58 @@ where
 	B: BlockT,
 	Client: HeaderBackend<B> + ProofProvider<B> + 'static,
 {
-	///  Create a new instance.
-	pub fn new(client: Arc<Client>, warp_sync_provider: Arc<dyn WarpSyncProvider<B>>) -> Self {
+	///  Create a new instance. When passing a warp sync provider we will be checking for proof and
+	/// authorities. Alternatively we can pass a target block when we want to skip downloading
+	/// proofs, in this case we will continue polling until the target block is known.
+	pub fn new(client: Arc<Client>, warp_sync_params: WarpSyncParams<B>) -> Self {
 		let last_hash = client.hash(Zero::zero()).unwrap().expect("Genesis header always exists");
-		let phase = Phase::WarpProof {
-			set_id: 0,
-			authorities: warp_sync_provider.current_authorities(),
-			last_hash,
+		match warp_sync_params {
+			WarpSyncParams::WithProvider(warp_sync_provider) => {
+				let phase = Phase::WarpProof {
+					set_id: 0,
+					authorities: warp_sync_provider.current_authorities(),
+					last_hash,
+					warp_sync_provider: warp_sync_provider.clone(),
+				};
+				Self { client, phase, total_proof_bytes: 0 }
+			},
+			WarpSyncParams::WaitForTarget(block) => Self {
+				client,
+				phase: Phase::PendingTargetBlock { target_block: block },
+				total_proof_bytes: 0,
+			},
+		}
+	}
+
+	/// Poll to make progress.
+	///
+	/// This only makes progress when `phase = Phase::PendingTargetBlock` and the pending block was
+	/// send.
+	pub fn poll(&mut self, cx: &mut std::task::Context) -> Poll<()> {
+		let new_phase = if let Phase::PendingTargetBlock { target_block } = &mut self.phase {
+			if let Poll::Ready(Ok(target_block)) = target_block.poll_unpin(cx) {
+				Some(Phase::TargetBlock(target_block))
+			} else if let Poll::Ready(Err(e)) = target_block.poll_unpin(cx) {
+				error!(target: "warp_sync", "Failed to get target block. Error: {:?}",e);
```

Comment:
```suggestion
				error!(target: "sync", "Failed to get target block. Error: {:?}",e);
```
This target is used by all of the sync logging